### DISCUSSION TITLE
Newbie Learning Materials

### DISCUSSION THREAD

**Discussion by DaviRain-Su** [informalsystems/hermes#1987] [Q&A] [answered]
Hi, is there any information for newbies to learn `ibc` and how to contribute to `ibc-rs`.

**Comment by adizere** [comment, marked as answer]
Good question! Our repo docs provide a good starting place when it comes to the Rust implementation:

- [Contributing.md](https://github.com/informalsystems/ibc-rs/blob/master/CONTRIBUTING.md)
- [main readme](https://github.com/informalsystems/ibc-rs#ibc-rs)
- the [ADRs](https://github.com/informalsystems/ibc-rs/tree/master/docs/architecture) provide rationale on various architectural decisions, e.g., [ADR-001 discusses repository structure](https://github.com/informalsystems/ibc-rs/blob/master/docs/architecture/adr-001-repo.md).


Further resources:

- Map of zone shows a status of IBC-enabled zones: https://mapofzones.com/?period=24
- the [cosmos/ibc](https://github.com/cosmos/ibc) repo provides pseudocode and English specifications of all IBC protocols e.g., [see this paper for an overview](https://github.com/cosmos/ibc/blob/master/archive/papers/2020-05/build/paper.pdf)

@andynog may know more about newbie material.

Did you have a specific kind of media in mind (e.g., video as opposed to text/papers)?

**Comment by DaviRain-Su** [comment]
> Did you have a specific kind of media in mind (e.g., video as opposed to text/papers)?

Yes it is better to provide this information. Also how does a newbie go about building a network using `ibc-rs` is there such information, substrate's website has similar information for newbies to get started, not sure if ibc has similar information.

**Comment by adizere** [comment]
I will let @andynog complement this thread with additional resources, in particular videos.

With respect to building a network on top of `ibc-rs`, there are two resources I can point you to:
- one example of a network that builds on top of `ibc-rs`, called [tendermock](https://github.com/CharlyCst/tendermock), with a slightly [more up-to-date fork here](https://github.com/adizere/tendermock), but neither are currently maintained
- we are building our own application around `ibc-rs`, which is in preliminary stages. @hu55a1n1 may have more context on this effort, and you can see the plan here https://gist.github.com/hu55a1n1/7aefee699ff4edc002403ebeaab7b6b5
